Docs: Explain how to enable experimental PromQL functions (#10712)
grafana/mimir: docs/sources/mimir/configure/configure-experimental-promql-functions.md

---
title: Configure experimental PromQL functions
description: Control what experimental PromQL functions are enabled for your Mimir installation.
weight: 110
---

# Configure experimental PromQL functions

> Experimental PromQL functions are disabled by default in Mimir. They can be selectively enabled per-tenant or globally for all users.

Reviewer: [nit] Could we list or link to a list of experimental PromQL functions?

Might also be useful to include the error message that is returned when an experimental PromQL function is used so that someone searching for it sees this page.

Author: > [nit] Could we list or link to a list of experimental PromQL functions?

Yep, but there's the danger of it going out of date. Not sure how much of an issue that is in practice. 

> Might also be useful to include the error message that is returned when an experimental PromQL function is used so that someone searching for it sees this page.

Good idea!

> Experimental PromQL functions are disabled by default in Mimir. They can be selectively enabled per-tenant or globally for all users.

Reviewer 2: ```suggestion
Experimental PromQL functions are disabled by default in Mimir. You can selectively enable them per-tenant or globally for all users.
```

Reviewer 2: We can use active voice here.

> All functions can be enabled using the special function name `all` or each individual function can be enabled specifically.

Reviewer 2: ```suggestion
You can enable all functions using the special function name `all`, or you can specifically enable individual functions.
```

You can enable functions globally for all users using the main Mimir configuration file.

```yaml
limits:
  enabled_promql_experimental_functions: info,sort_by_label,sort_by_label_desc
```

> You can enable functions for specific tenants using [per-tenant overrides]({{< relref "./about-runtime-configuration" >}}):

Reviewer 2: We should use the full URL here instead of a relref.

Author: I'm going to leave this as-is in this PR since it seems like all the other docs in `configure/` are using relref. We can update them all at once.

```yaml
overrides:
  "tenant-id-1":
    # Enable only specific functions for this tenant
    enabled_promql_experimental_functions: sort_by_label,sort_by_label_desc
  "tenant-id-2":
    # Enable all experimental functions for this tenant
    enabled_promql_experimental_functions: all
```

> Blocking or allowing use of experimental PromQL functions is enforced on instant and range queries.

Reviewer 2: ```suggestion
Blocking or allowing the use of experimental PromQL functions is enforced on instant and range queries.
```
